jupyter_core/command.py

```diff
@@ -116,8 +117,10 @@ def main():
     command = 'jupyter-' + subcommand
     try:
         os.execvp(command, sys.argv[1:])
-    except OSError:
-        sys.exit("jupyter: %r is not a Jupyter command" % subcommand)
+    except OSError as e:
+        if e.errno == errno.ENOENT:
+            sys.exit("jupyter: %r is not a Jupyter command" % subcommand)
+        raise
```

Comment:
Okay. I think the error message but no traceback would be better, but this is fine.